src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java

```diff
@@ -202,7 +203,18 @@ private StartWorkflowExecutionParameters getStartWorkflowExecutionParameters(
     }
     Map<String, byte[]> result = new HashMap<>();
     for (ContextPropagator propagator : contextPropagators) {
-      result.putAll(propagator.serializeContext(propagator.getCurrentContext()));
+      // Get the serialized context from the propagator
+      Map<String, byte[]> serializedContext =
+          propagator.serializeContext(propagator.getCurrentContext());
+      // Namespace each entry in case of overlaps, so foo -> bar becomes MyPropagator:foo -> bar
+      Map<String, byte[]> namespacedSerializedContext =
+          serializedContext
+              .entrySet()
+              .stream()
+              .collect(
+                  Collectors.toMap(
+                      k -> propagator.getName() + ":" + k.getKey(), Map.Entry::getValue));
+      result.putAll(namespacedSerializedContext);
```

Comment:
maybe move to a util to reuse the code. It's repeated 4 times for ChildWF, WF, activity and localActivity.